src/blockdata/script.rs

```diff
@@ -325,6 +325,26 @@ impl Script {
                                opcodes::All::from(self.0[0]).classify() == opcodes::Class::IllegalOp)
     }
 
+    /// The minimum value an output to a witness script must have in order to be
+    /// broadcastable on today's bitcoin network.
+    pub const WITNESS_OUTPUT_DUST_THRESHOLD: u64 = 294;
+
+    /// The minimum value an output to a non-witness script must have in order to be
+    /// broadcastable on today's bitcoin network.
+    pub const LEGACY_OUTPUT_DUST_THRESHOLD: u64 = 546;
+
+    /// Gets the minimum value an output with this script should have in order to be
+    /// broadcastable on today's bitcoin network.
+    pub fn dust_value(&self) -> u64 {
+        if self.is_op_return() {
```

Comment:
When looking [at core](https://github.com/bitcoin/bitcoin/blob/6c6140846f37de8c132b3b6abf09f3d7940554a7/src/policy/policy.cpp#L30) it also captures the [case where a script is too big](https://github.com/bitcoin/bitcoin/blob/6c6140846f37de8c132b3b6abf09f3d7940554a7/src/script/script.h#L545) making the UTXO unspendable. But that would only make this implementation more conservative which might be ok since it's such a fringe use case and not consensus critical.

Reply:
Hmm, right - are those scripts broadcastable at all or are they nonstandard?

Reply:
Those scripts are not possible by consensus rules all scripts(bare, p2sh, witness_v0). But post tapscript, those scripts will be broadcastable. 

https://github.com/bitcoin/bips/blob/master/bip-0342.mediawiki#resource-limits

https://github.com/bitcoin/bitcoin/blob/6c6140846f37de8c132b3b6abf09f3d7940554a7/src/script/interpreter.cpp#L452

We can change the isWitnessProgram check to check only v0 programs, but this can also be done in taproot upgrade #503. But this could have easily been one of the things we forgot to add.

Reply:
@sgeisler  Nevermind, I was wrong. This function is about scriptPubkey and not about the witness Program inside it :).

Reply:
Right, I'm not even talking about if its *valid* (its obviously not possible to spend), but is it standard - can you broadcast a transaction with such a script in an output and have it be mined, even if the output is unspendable.